ClientInfo.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace DNWS
+{
+  class ClientInfo : IPlugin
```

Comment:
You should follow the pattern, put "Plugin" as suffix to the class name, so use can know that this is a plugin class.